dvc/stage/monitor.py

```diff
@@ -36,22 +37,141 @@ class MonitorTask:
     def name(self) -> str:
         raise NotImplementedError
 
-    @property
-    def SIGNAL_FILE(self) -> str:
-        raise NotImplementedError
-
     @property
     def error_cls(self) -> type:
         raise NotImplementedError
 
-    @property
-    def signal_path(self) -> str:
-        return os.path.join(self.stage.repo.tmp_dir, self.SIGNAL_FILE)
+    def is_ready(self):
+        pass
+
+    def on_start(self):
+        pass
+
+    def on_call_end(self):
+        pass
 
-    def after_run(self):
+    def on_end(self):
         pass
 
 
+class LiveStudioTask(MonitorTask):
+    name = "live_studio"
+    error_cls = LiveStudioKilledError
+
+    def __init__(self, stage: "Stage", proc: subprocess.Popen, url: str):
+        super().__init__(stage, self._send_updates, proc)
+        import requests
+        from requests.adapters import HTTPAdapter, Retry
+
+        self.fs = stage.repo.fs
+        self.url = url
+        self.rev = stage.repo.scm.get_rev()
+
+        self.session = requests.Session()
+        self.session.headers.update({"content-type": "application/json"})
+        retries = Retry(
```

Comment:
A more fancy way to retry would be to track what failed and if a new event comes join them and send together. E.g. do join if a new event comes before the timeout for retry ends. May affect the format.

That's optional of cause, more of a thought.